### ISSUE TITLE
Support introspection queries on GraphQL endpoint

### ISSUE THREAD

**Issue by ra0x3** [FuelLabs/fuel-indexer#790]
- This is a feature request from @luizstacio 
- Our GraphQL endpoint currently supports translating GraphQL queries to SQL queries
- However, that is not all this endpoint should do
- The endpoint should also determine whether or not the given request is for an "introspection" query
- If the request is for an "introspection" query, the endpoint should _not_ try to translate that to SQL
  - The endpoint should just parse the GraphQL schema, and return the introspection data
- This helps the FE team productivity by auto-completing queries and types, generating docs, and generating schema as well

Additional context
- Currently the `fuel-core` playground supports this
- If you open the fuel-core playground http://beta-3.fuel.network/playground and start typing a query, you can see the result of the introspection query
- Example output:
[sample-introspection-query.txt](https://github.com/FuelLabs/fuel-indexer/files/11312486/sample-introspection-query.txt)

**Comment by rvmelkonian** [comment]
Yes, I thought this one would come up.. I think this is a big value add for users.
